Bootstrap v4.0.0-alpha1

sebakerckhof opened the pull request:
Initial bootstrap v4.0.0 support (based on alpha 1 docs).

In general following has changed in bootstrap 4 and is reflected here:
- navbar <li> elements need a nav-item class
- navbar <a> elements need a nav-link class
- dropdown content is no longer a list, but are directly links
- dropdown headers have an h6-tag
- glyphicons are removed

sebakerckhof committed:
Bootstrap v4.0.0-alpha1

pattonwebz commented:
Nice one for jumping on this @sebakerckhof, never too early to get started on an updated version :)

I do remember that with BS3 release there were several changes to nav elements in a few releases between the first alpha and final release so let's keep an eye on this as v4 evolves.

I'll start a new branch for the v4 version of the walker tomorrow and test this out.

pattonwebz commented:
This item: 
- dropdown headers have an h6-tag 

Do you know if they NEED to be `<h6>` tags? Are they possibly just `<h6>` tags as a method to visually distinguish them from normal nav items and in reality they could use just css targeting to accomplish that?

sebakerckhof commented:
The styling is indeed applied only by class-name, not by tag-name, but in the [docs](http://v4-alpha.getbootstrap.com/components/dropdowns/#menu-headers) they used h6's, which I tried to follow as close as possible

pattonwebz commented:
Ahh OK I understand now, that's exactly what I expected but wanted to clarify - I hadn't yet looked at the docs for v4 closely as I've been busy with work the last week.

Also @twittem is now working directly on a version for v4 so I expect that your work and his could help each other during his first build. Ed, have you looked at this PR?

sebakerckhof committed:
dropdown tag fix

sebakerckhof commented:
Quick note, I forgot to change the level tags (dropdown menu outer tags). This has now been fixed in the latest commit.

danimalweb commented:
Great work on this.

Will we get a new branch soon? I'm using BS4 in a new project and had to write a bunch of extra CSS to abstract the class names to get it to work with the current master branch. Thanks.

ci2014 committed:
Update wp_bootstrap_navwalker.php

In Line 58 there was if(strcasecmp( $item->attr_title, 'divider' ) == 0 || strcasecmp( $item->title, 'divider') == 0)) instead of if(strcasecmp( $item->attr_title, 'divider' ) == 0 || strcasecmp( $item->title, 'divider') == 0)

sebakerckhof committed:
Merge pull request #1 from ci2014/patch-1

SideQuestDIY commented:
Sorry for the delay on this and thanks for the work on it. I am currently restructuring the walker to have a pluggable nature for adding additional functionality. I am hoping to have the dev branch up by November 9th.

pattonwebz commented:
November 9th isn't all that far away. I did hack up something from the 
old walker that works for BS4 that I've been using in a recent project 
but will be happy once there's a proper v4 version here :)

Let me know if I could be any help with building this latest dev branch.

On 03/11/15 17:30, Edward McIntyre wrote:

> Sorry for the delay on this and thanks for the work on it. I am 
> currently restructuring the walker to have a pluggable nature for 
> adding additional functionality. I am hoping to have the dev branch up 
> by November 9th.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/twittem/wp-bootstrap-navwalker/pull/190#issuecomment-153425972.

mindctrl commented:
jQuery 2 requirement might be an issue. Haven't looked at BS 4 at all yet though. https://github.com/twbs/bootstrap/pull/17494

ghost commented:
Also looking forward to v4 support. Hopefully we would be able to create a three level menu too.

PascaleBeier commented:
Just wanted to add that bootstrap requires jQuery >= 1.9.1 and != jQuery 3. I'm on mobile so I can't link it - but you find it in the latest commits.

mindctrl commented:
https://github.com/twbs/bootstrap/pull/18267

jordansaspect commented:
I am having trouble outputting the nav-item class on the list items

jamesfacts commented:
This was very helpful—thanks @sebakerckhof

leqnam commented:
very helpful but how the progress is going on?

bhubbard commented:
The navwalker has gone thru many changes for 2.0.5, and we are gonna use that for the basis of support for Bootstrap 4. It is something planned for the roadmap, but if you want to take another shot at it feel free to submit a new pull request.
